woocommerce/woocommerce-ios
# My Store: Fix incorrect currency code in revenue text after syncing store settings the first time

Closes: #6301


### Description

When first opening the app or switching to a new store, the app needs to sync store settings before updating currency settings. The default currency code is USD and the currency symbol is $, and the revenue text is created before the syncing completes. So if the store uses any currency other than the default, the revenue text on the My Store screen will be incorrect after the syncing completes.

The solution is to update revenue text when currency code changes:
- Inject currency settings to `StoreStatsPeriodViewModel` instead of currency code.
- Update the `currencyCode` property in `CurrencySettings` to make it a `@Published` property.
- Combine the changes of order stats, selected time intervals, and the currency code to update revenue text accordingly.

### Testing instructions

- Make sure to have a test store with a currency symbol other than $.
- Switch to that store on the app if needed.
- Relaunch the app, make sure that the currency symbol displayed on the revenue text on the My Store page is correct immediately.

### Screenshots

Before:
https://user-images.githubusercontent.com/5533851/156137323-33f18cb9-928c-4158-886f-17df44b9defc.mp4


After:
https://user-images.githubusercontent.com/5533851/156137378-20d0fcae-3cda-4682-9d0f-77a31232e43a.mp4

---
